Application/Business/AttributeGroup/AttributeGroupServices.cs

```diff
@@ -36,20 +37,33 @@ public async Task Create(AttributeGroupEntity attributeGroup)
         }
         if (!check)
         {
-            var attGrp = new AttributeGroupEntity
-            {
-                Name = attributeGroup.Name,
-                Effect = attributeGroup.Effect
-            };
-            await _attributeRepo.CreateAsync(attGrp);
+            await _attributeRepo.CreateAsync(attributeGroup);
+        }
+        else
+        {
+            throw new BadRequestException("Name already exist.");
         }
     }
     public async Task Update(Guid attributeGroupid, AttributeGroupEntity attributeGroup)
     {
-
+        var target = await GetById(attributeGroupid);
+        if(target is not null)
+        {
+            target.Name = attributeGroup.Name;
+            target.Effect = attributeGroup.Effect;
+            await _attributeRepo.UpdateAsync(target);
+        }
+        else
+        {
+            throw new NotFoundException("Attribute group not exist");
+        }
     }
     public async Task Delete(Guid attributeGroupid)
     {
-
+        var target = await _attributeRepo.DeleteSoftAsync(attributeGroupid);
```

Comment:
check this function logic, why do you delete before checking if this entity exists???

Reply:
The function DeleteSoftAsync changes DeletedAt and returns item if it exist. 

```
public virtual async Task<T> DeleteSoftAsync(Guid id)
    {
        T _entity = await FindByIdAsync(id);
        if (_entity == null)
        {
            return null;
        }
        _entity.DeletedAt = DateTime.UtcNow;
        await UpdateAsync(_entity);
        return _entity;
    }
```

Reply:
finding non-entity is user error, return error to client. You have to check whether the entity exists or not before performing the logic